runner/internal/shim/docker.go

```diff
@@ -595,7 +595,13 @@ func requestGpuIfAvailable(ctx context.Context, client docker.APIClient) ([]cont
 	for runtime := range info.Runtimes {
 		if runtime == consts.NVIDIA_RUNTIME {
 			return []container.DeviceRequest{
-				{Capabilities: [][]string{{"gpu"}}, Count: -1}, // --gpus=all
+				{
+					// Request all capabilities to maximize compatibility with all sorts of GPU workloads.
+					// Default capabilities: utility, compute.
+					// https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/1.16.0/docker-specialized.html
+					Capabilities: [][]string{{"gpu", "utility", "compute", "graphics", "video", "display", "compat32"}},
```

Comment:
(nit) Consider using [`"all"`](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/1.16.0/docker-specialized.html?highlight=all#driver-capabilities) if it's allowed here

Reply:
It' s only works with `NVIDIA_DRIVER_CAPABILITIES` env var, not CLI (`--gpus 'all,"capabilities=all"'` won't work), not API